src/App.tsx

```diff
@@ -1,8 +1,61 @@
-import React from 'react';
+import React, { useState, FC } from 'react';
 import './App.css';
+import { TodosList } from './components/TodosList/TodosList';
+import { getUsers, getTodos } from './api';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App: FC<{}> = () => {
+  const [isLoaded, setLoaded] = useState<boolean>(false);
+  const [todos, setTodos] = useState<TodoWithUser[]>([]);
+
+  const loadTodos = async () => {
+    setLoaded(true);
+    const [todosLoaded, users] = await Promise.all([getTodos(), getUsers()]);
+
+    const todosWithUsers = todosLoaded.map(todo => ({
+      ...todo,
+      user: users.find(user => user.id === todo.userId),
+    }));
+
+    setTodos(todosWithUsers);
+  };
+
+  const sortByTitle = () => {
+    setTodos([...todos].sort((a, b) => a.title.localeCompare(b.title)));
+  };
+
+  const sortByName = () => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    setTodos([...todos].sort((a, b) => a.user!.name.localeCompare(b.user!.name)));
+  };
+
+  const sortByStatus = () => {
+    setTodos([...todos].sort((a, b) => Number(b.completed) - Number(a.completed)));
+  };
+
+  if (todos.length === 0) {
+    return (
+      <>
+        <button
+          type="button"
+          className="btn-load"
+          onClick={loadTodos}
+          disabled={isLoaded}
+        >
+          Load
+        </button>
+        <p className="text">{isLoaded ? 'Loading...' : ''}</p>
```

Comment:
Зачем нам пустой `<p></p>`?
```suggestion
        {isLoaded && (
          <p className="text">Loading...</p>
        )}
```

Reply:
как я понял, это процесс загрузки, тогда `isLoading`

Reply:
Changed